### PR TITLE
Update setup.py to get version info from git tags.

### PR DESCRIPTION
**Pull request by maffoo**
We make some much-needed updates of setup.py for pylabrad. In particular, we add support for the `setuptools_scm` package which pulls version information from git tags when building a distribution package. This should address the concerns around versioning in #203. Of course, we still have to tag versions in git and upload packages to PyPI, but modern tools make this fairly easy. For reference, here is the official guide to python packaging: https://python-packaging.readthedocs.org/en/latest/

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,2 +1,2 @@
-include *.txt
-include *.ini
```
Comment (DanielSank) [inline MANIFEST.in]: These two lines didn't actually change. Remove this "change" from the commit?

**Comment by DanielSank** [comment]
LGTM (but maybe fix that little weird change to `MANIFEST.in`)

File: MANIFEST.in (hunk 1 above)
Comment (maffoo) [inline MANIFEST.in]: I changed the line endings to UNIX. git will check them out as CRLF on windows if that's how people have it configured.

**Comment by maffoo** [comment]
@DanielSank, thanks for looking at this. I just added some scripts to build and upload new packages; if you could take another look that'd be great.

Once this is in, I'd like to tag and push a new version, we just need to decide what to bump the version number up to. 0.94.0? 1.0.0? Would be glad to hear your thoughts.

File: MANIFEST.in (hunk 1 above)
Comment (DanielSank) [inline MANIFEST.in]: Ah, very good.

**Comment by DanielSank** [comment]
Linux
* `dist_build.sh`
* `dist_upload.sh`

Windows
* `make_dist.bat`
* `pypi_upload.bat`

It might be nice to use parallel structure in the file names.

**Comment by maffoo** [comment]
I actually meant to delete those windows scripts... But I can change them. The only reason for building on windows too is if we want to have windows installers (instead of just wheels) and/or if we ever decide to include c-extensions. I've been pushing back on c-extensions precisely because building packages for all platforms is kind of a pain.

**Comment by DanielSank** [comment]
Ok, let's either rename or delete. I kinda vote for delete.

**Comment by maffoo** [comment]
oh, I did delete them. You might just need to refresh your diff on github.

**Comment by DanielSank** [comment]
A quick note in the readme to the effect of "To build and update PyPI do blah blah blah" wouldn't hurt. Something like 1) Tag a commit, 2) Build the distribution, 3) Update PyPI. But anyway, LGTM.

**Pull request merged**